### PR TITLE
EventHub - Swagger checklist review comments updated

### PR DESCRIPTION
**Pull request by v-Ajnava**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [X] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [X] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [X] My spec meets the review criteria:
  - [X] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [X] Validation errors from the [Linter extension for VS Code](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/linter.md) have all been fixed for this spec. (**Note:** for large, previously checked in specs, there will likely be many errors shown. Please contact our team so we can set a timeframe for fixing these errors if your PR is not going to address them).
  - [X] The spec follows the patterns described in the [Swagger good patterns](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-good-patterns.md) document unless the service API makes this impossible.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@v-Ajnava__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (v-ajnava). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: arm-eventhub/2015-08-01/swagger/EventHub.json [hunk 1]
```diff
@@ -780,35 +858,43 @@
...
+            "maxLength": 50,
+            "description": "Authorization rule name."
           },
           {
             "name": "parameters",
             "in": "body",
             "required": true,
             "schema": {
-              "$ref": "#/definitions/SharedAccessAuthorizationRuleCreateOrUpdateParameters"
+              "$ref": "#/definitions/SharedAccessAuthorizationRuleResource"
```
Comment (anudeepsharma) [inline arm-eventhub/2015-08-01/swagger/EventHub.json]: SharedAccessAuthorizationRuleResource can be renamed to SharedAccessAuthorizationRule

File: arm-eventhub/2015-08-01/swagger/EventHub.json [hunk 2]
```diff
@@ -1074,35 +1192,43 @@
...
           },
           {
             "name": "namespaceName",
             "in": "path",
             "required": true,
             "type": "string",
-            "description": "The namespace name."
+            "minLength": 6,
+            "maxLength": 50,
+            "description": "The Namespace Name within the Resource Group"
```
Comment (anudeepsharma) [inline arm-eventhub/2015-08-01/swagger/EventHub.json]: The name of the Namespace (with in the resource group can be removed, as even outside it is same)

File: arm-eventhub/2015-08-01/swagger/EventHub.json [hunk 3]
```diff
@@ -112,21 +112,25 @@
...
           },
           {
             "name": "parameters",
             "in": "body",
             "required": true,
             "schema": {
-              "$ref": "#/definitions/NamespaceCreateOrUpdateParameters"
+              "$ref": "#/definitions/NamespaceResource"
```
Comment (sazeesha) [inline arm-eventhub/2015-08-01/swagger/EventHub.json]: Should this be renamed to Namespace?

File: arm-eventhub/2015-08-01/swagger/EventHub.json [hunk 4]
```diff
@@ -534,14 +583,18 @@
...
           {
             "name": "namespaceName",
             "in": "path",
             "required": true,
             "type": "string",
-            "description": "The namespace name."
+            "minLength": 6,
+            "maxLength": 50,
+            "description": "The Namespace Name within the Resource Group"
           },
```
Comment (sazeesha) [inline arm-eventhub/2015-08-01/swagger/EventHub.json]: Please change this description to "The name of the namespace"

File: arm-eventhub/2015-08-01/swagger/EventHub.json [hunk 5]
```diff
@@ -579,28 +632,34 @@
...
           {
             "name": "namespaceName",
             "in": "path",
             "required": true,
             "type": "string",
-            "description": "The namespace name."
+            "minLength": 6,
+            "maxLength": 50,
+            "description": "The Namespace Name within the Resource Group"
```
Comment (sazeesha) [inline arm-eventhub/2015-08-01/swagger/EventHub.json]: Please change this description to "The name of the namespace"

Comment (v-Ajnava) [inline arm-eventhub/2015-08-01/swagger/EventHub.json]: Resolved.

File: arm-eventhub/2015-08-01/swagger/EventHub.json (hunk 2 above)
Comment (v-Ajnava) [inline arm-eventhub/2015-08-01/swagger/EventHub.json]: Resolved, corrected with 'The name of the namespace'

File: arm-eventhub/2015-08-01/swagger/EventHub.json (hunk 1 above)
Comment (v-Ajnava) [inline arm-eventhub/2015-08-01/swagger/EventHub.json]: Resolved.

File: arm-eventhub/2015-08-01/swagger/EventHub.json [hunk 6]
```diff
@@ -1550,22 +1624,27 @@
...
         },
         "primaryKey": {
           "type": "string",
+          "readOnly": true,
           "description": "A base64-encoded 256-bit primary key for signing and validating the SAS token."
         },
         "secondaryKey": {
           "type": "string",
+          "readOnly": true,
           "description": "A base64-encoded 256-bit primary key for signing and validating the SAS token."
```
Comment (v-Ajnava) [inline arm-eventhub/2015-08-01/swagger/EventHub.json]: Resolved, changed to Secondary.

File: arm-eventhub/2015-08-01/swagger/EventHub.json [hunk 7]
```diff
@@ -112,21 +112,25 @@
             "in": "path",
```
Comment (v-Ajnava) [inline arm-eventhub/2015-08-01/swagger/EventHub.json]: Resolved.

File: arm-eventhub/2015-08-01/swagger/EventHub.json (hunk 3 above)
Comment (anudeepsharma) [inline arm-eventhub/2015-08-01/swagger/EventHub.json]: yes

**Review by anuchandy — COMMENTED** [review]
In what format metadata is stored in ConsumerGroupProperties::userMetadata property (xml, json, csv) ? Can we add this info to description if applicable https://github.com/v-Ajnava/azure-rest-api-specs/blob/59ed43fc1e362a79bdf7a304178f25b9640000a9/arm-eventhub/2015-08-01/swagger/EventHub.json#L1336

File: arm-eventhub/2015-08-01/swagger/EventHub.json (hunk 3 above)
Comment (v-Ajnava) [inline arm-eventhub/2015-08-01/swagger/EventHub.json]: Resolved.

File: arm-eventhub/2015-08-01/swagger/EventHub.json [hunk 8]
```diff
@@ -41,7 +39,7 @@
         "tags": [
           "Namespaces"
```
Comment (anuchandy) [inline arm-eventhub/2015-08-01/swagger/EventHub.json]: Does eventhub RP supports API to check the availability of namespace? if yes can we add it?

**Review by anuchandy — COMMENTED** [review]
Is EventHubProperties::status a read-only property then please marked so.

File: arm-eventhub/2015-08-01/swagger/EventHub.json (hunk 8 above)
Comment (v-Ajnava) [inline arm-eventhub/2015-08-01/swagger/EventHub.json]: Resolved, included the API.

File: arm-eventhub/2015-08-01/swagger/EventHub.json [hunk 9]
```diff
@@ -1752,6 +1698,86 @@
...
+          },
+          "description": "Resource tags"
+        },
+        "sku": {
+          "$ref": "#/definitions/Sku",
+          "description": "The sku of the created namespace"
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/NamespaceProperties"
```
Comment (anuchandy) [inline arm-eventhub/2015-08-01/swagger/EventHub.json]: All the properties of NamespaceProperties [provisioningState, createdAt, updatedAt, serviceBusEndpoint and metricId] seems eligible to mark as read-only. Could you please check? If all are read-only why namespace PATCH model (NamespacePatch) includes NamespaceProperties?

File: arm-eventhub/2015-08-01/swagger/EventHub.json [hunk 10]
```diff
@@ -1752,6 +1698,86 @@
...
+        },
+        "message": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The detailed info regarding the reason associated with the namespace."
+        }
+      },
+      "description": "Description of a Check Name availability request properties."
+    },
+    "NamespacePatch": {
```
Comment (anuchandy) [inline arm-eventhub/2015-08-01/swagger/EventHub.json]: better name for example, NamespacePatch -> NamespaceUpdateParameters

File: arm-eventhub/2015-08-01/swagger/EventHub.json (hunk 9 above)
Comment (v-Ajnava) [inline arm-eventhub/2015-08-01/swagger/EventHub.json]: Resolved

File: arm-eventhub/2015-08-01/swagger/EventHub.json (hunk 10 above)
Comment (v-Ajnava) [inline arm-eventhub/2015-08-01/swagger/EventHub.json]: Resolved

**Review by anudeepsharma — APPROVED** [review]
Looks good to me

**Pull request merged**